### PR TITLE
Add IFEU holidays

### PR DESCRIPTION
**Pull request by benjfield**
## Proposed change

I have added the base IFEU holidays which are used both for the exchange and for settlement pricing for products including Brent Futures and Low Sulphur Gasoil Futures.

## Type of change

- [x] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

### CODE REVIEW

**Review by arkid15r — COMMENTED** [review]
Hi @benjfield 

Thanks for the clean code PR! I don't have much suggestions after glancing the PR. 
Apparently, we need to update the supported markets [section of the README.rst fle](https://github.com/vacanza/python-holidays/blob/353d86df95c2d10207ed9dfaeaea31dd923c22ee/README.rst?plain=1#L925)

I'll take another look later today.

Thanks again for your contribution!

File: tests/financial/test_ice_futures_europe.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(ICEFuturesEurope(years=2013))
+
+    def test_2021(self):
+        self.assertHolidays(
+            ICEFuturesEurope(years=2021),
+            ("2021-01-01", "New Year's Day"),
+            ("2021-04-02", "Good Friday"),
+            # Saturday so not a market holiday
```
Comment (arkid15r) [inline tests/financial/test_ice_futures_europe.py:35]: I'm not sure if the comment is useful here. Could you explain what you meant by it? It seems a bit controversial within the `assertHolidays` context.

Thank you!

**Review by arkid15r — COMMENTED** [review]
Could you also include `IFEU` snapshot (see `make snapshot` command)?

Thank you!

File: tests/financial/test_ice_futures_europe.py (hunk 1 above)
Comment (arkid15r) [reply tests/financial/test_ice_futures_europe.py]: I see, thanks for the explanation. 
I've implemented https://github.com/vacanza/python-holidays/pull/1796 today in order to improve this. Now we have `SAT_TO_NONE` rule. 

I also updated the README.

I believe it's good to go.

**Review by arkid15r — COMMENTED** [review]
LGTM

Any suggestions @kjhellico

**Review by KJhellico — APPROVED** [review]
LGTM!

**Pull request merged**